pennylane/dla/lie_closure.py

```diff
@@ -24,6 +25,61 @@
 from pennylane.pauli import PauliSentence, PauliWord
 
 
+def lie_closure(
+    generators: Iterable[Union[PauliWord, PauliSentence, Operator]],
+    max_iterations: int = 1e4,
+    verbose: int = 0,
+) -> Iterable[Union[PauliWord, PauliSentence, Operator]]:
+    r"""Compute the Lie closure over commutation of a set of generators
+
+    The Lie closure of a set of generators :math:`\mathcal{G} = \{G_1, .. , G_N\}` is computed by
+    taking all possible nested commutators until no new (i.e. linearly independent) operator is produced.
+
+    Args:
+        generators (Iterable[Union[PauliWord, PauliSentence, Operator]]): generating set for which to compute the
+            Lie closure.
+        max_iterations (int): maximum depth of nested commutators to consider.
+
+    Returns:
+        list[`~.PauliSentence`]: a basis of ``PauliSentence`` instances that is closed under
+        commutators (Lie closure).
+    """
+    if not all(isinstance(op, PauliSentence) for op in generators):
+        generators = [
+            op.pauli_rep if op.pauli_rep else qml.pauli.pauli_sentence(op) for op in generators
+        ]
+
+    vspace = PauliVSpace(generators)
+
+    epoch = 0
+    old_length = 0  # dummy value
+    new_length = len(vspace)
+
+    while new_length > old_length or epoch > max_iterations:
```

Comment:
The dimension can blow up to up to $4^n-1$ for $n$ qubits, but some DLAs (i.e. reps of $su(2^n)$ ) are just that so I wouldnt force an arbitrary threshold below that (and it cant go beyond)

We could have an additional, optional, hyper parameter the user can set `max_size` or so?

Reply:
I think you know better what is best here. We could just let it as is, throw the feature into the wild, and see whether this becomes a problem in production. Maybe it will bring some interesting DLAs to our attention :)

Reply:
I would compare this to users trying to initialize a state vector with too many qubits. The object simply is very large in some cases, and an incomplete DLA, as @Qottmann points out, would not serve too much of a purpose here :)